internal/service/project/data_source_project.go

```diff
@@ -156,28 +155,28 @@ func (d *projectDS) Read(ctx context.Context, req datasource.ReadRequest, resp *
 
 	var (
 		err     error
-		project *matlas.Project
+		project *admin.Group
 	)
 
 	if !projectState.ProjectID.IsNull() {
 		projectID := projectState.ProjectID.ValueString()
-		project, _, err = conn.Projects.GetOneProject(ctx, projectID)
+		project, _, err = connV2.ProjectsApi.GetProject(ctx, projectID).Execute()
 		if err != nil {
 			resp.Diagnostics.AddError("error when getting project from Atlas", fmt.Sprintf(ErrorProjectRead, projectID, err.Error()))
 			return
 		}
 	} else {
 		name := projectState.Name.ValueString()
-		project, _, err = conn.Projects.GetOneProjectByName(ctx, name)
+		project, _, err = connV2.ProjectsApi.GetProjectByName(ctx, name).Execute()
 		if err != nil {
 			resp.Diagnostics.AddError("error when getting project from Atlas", fmt.Sprintf(ErrorProjectRead, name, err.Error()))
 			return
 		}
 	}
 
-	atlasTeams, atlasLimits, atlasProjectSettings, err := getProjectPropsFromAPI(ctx, conn, connV2, project.ID)
+	atlasTeams, atlasLimits, atlasProjectSettings, err := getProjectPropsFromAPI(ctx, connV2, *project.Id)
 	if err != nil {
-		resp.Diagnostics.AddError("error when getting project properties", fmt.Sprintf(ErrorProjectRead, project.ID, err.Error()))
+		resp.Diagnostics.AddError("error when getting project properties", fmt.Sprintf(ErrorProjectRead, *project.Id, err.Error()))
```

Comment:
```suggestion
		resp.Diagnostics.AddError("error when getting project properties", fmt.Sprintf(ErrorProjectRead, project.GetId(), err.Error()))
```
use the getters to avoid pointer operations

Reply:
Thanks! Done

Reply:
thx @gssbzn , this is something we're not doing consistently